torch_geometric/loader/neighbor_loader.py

```diff
@@ -121,14 +119,7 @@ def __init__(
                 set(edge_attr.edge_type for edge_attr in edge_attrs))
 
             # Set other required parameters:
-            if isinstance(num_neighbors, (list, tuple)):
-                num_neighbors = {key: num_neighbors for key in self.edge_types}
-            assert isinstance(num_neighbors, dict)
-            self.num_neighbors = {
-                edge_type_to_str(key): value
-                for key, value in num_neighbors.items()
-            }
-            self.num_hops = max([len(v) for v in self.num_neighbors.values()])
+            self._set_num_neighbors_num_hops(num_neighbors)
```

Comment:
Can you do this once outside of the condition instead of twice?

Reply:
Ah that would be nice. At the moment we can't because `Data` doesn't have `edge_types` and `_set_num_neighbors_num_hops` needs that. We could add another if condition to check if `edge_type` is an attribute and then call `_set_num_neighbors_num_hops`.